client/composables/tokens/consumers/useSharexConfiguration.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * This class represent the configuration file that is used to configure the *ShareX* app.
+ */
+
+class SharexConfiguration {
+  private Version = '15.0.0'
+  private Name: string
+  private DestinationType = 'ImageUploader, TextUploader, FileUploader'
+  private RequestMethod = 'POST'
+  private RequestURL: string
+  private Headers: object
+  private Body = 'FormURLEncoded'
+  private FileFormName = 'file'
+  private URL: string
+  private ThumbnailURL: string
+  private DeletionURL = ''
+  private ErrorMessage = ''
+
+  constructor(name: string, baseURL: string, token: string) {
+    this.Name = name
+    this.RequestURL = `${baseURL}/api/files/upload/token`
+    this.URL = `${baseURL}/{json:code}`
+    this.ThumbnailURL = `${baseURL}/{json:code}`
+    this.Headers = {
+      'Content-Type': 'application/x-www-form-urlencoded',
+      'Upload-Token': token,
+    }
+  }
+}
+
+export default function useSharexConfiguration() {
+  const baseURL = window.location.origin
```

Comment:
We need to keep in mind this will break when we decide to stop bundling the client app inside the server app

Reply:
Yes, I'm aware of this point of concern.